<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>BIOL 322 Lab 3. Genetic Drift • biol322</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet">
<meta property="og:title" content="BIOL 322 Lab 3. Genetic Drift">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-primary" data-bs-theme="dark" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">biol322</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../index.html"><span class="fa fa-home fa-lg"></span></a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-course-materials" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Course Materials</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-course-materials">
<li><hr class="dropdown-divider"></li>
    <li><a class="dropdown-item" href="../articles/sheets/sci_articles/biol322-syllabus-Roles-spring2026.pdf">Syllabus PDF</a></li>
    <li><a class="external-link dropdown-item" href="https://docs.google.com/document/d/1sV-864Y8L7gAb3LwBSqtKKL0P0aBKue20PB63_IPu-Y/edit?usp=sharing">Grade Contract Template</a></li>
    <li><a class="external-link dropdown-item" href="https://drive.google.com/drive/folders/1arcg7_nXRov4kTE3jgnS3q8DQ9DwZcFO?usp=sharing">Course GDrive folder</a></li>
    <li><a class="external-link dropdown-item" href="https://drive.google.com/drive/folders/1xDjUUTZY1j88zHregL34JMnaW6BZAUzq?usp=drive_link">In-class materials</a></li>
    <li><a class="dropdown-item" href="../articles/lab-exercises-main.html">Lab Exercises</a></li>
    <li><a class="external-link dropdown-item" href="https://drive.google.com/drive/folders/1ZuXCOFdWGMYW6JMkuYXSKr1I2bELQYDm?usp=drive_link">Exams</a></li>
    <li><a class="external-link dropdown-item" href="https://docs.google.com/document/d/1VkC-BHN8_EGvSv86qwt3uVwcK--VN61rOL4qlOGoAos/edit?usp=drive_link">Midterm Reflection</a></li>
    <li><a class="external-link dropdown-item" href="https://drive.google.com/drive/folders/18AjoufAqq3saqJmHq8bZrmXG0pGoEM00?usp=drive_link">Student Folders</a></li>
    <li><a class="external-link dropdown-item" href="https://docs.google.com/spreadsheets/d/1cfcKbMkjukYi1RewOfqhoLowUIsxJcsT34LbCcgj39I/edit?usp=drive_link">Check Progress Spreadsheet</a></li>
    <li><hr class="dropdown-divider"></li>
  </ul>
</li>
<li class="nav-item"><a class="external-link nav-link" href="https://docs.google.com/document/d/1gNcEPhAxowrrbwCOjgP2J_djOn_QOkaKBlhnRhSO5iQ/edit?usp=drive_link">Course Agenda</a></li>
<li class="nav-item"><a class="external-link nav-link" href="https://docs.google.com/forms/d/e/1FAIpQLSdA2mkUojv-37gaK0ON-B8B2a7y5cxnMUJRhaqX747QtbMxhA/viewform?usp=header">Assignment Turn In</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><a class="nav-link" href="../articles/sheets/sci_articles/biol322-syllabus-Roles-spring2026.pdf">BIOL 322 Spring 2026 Syllabus PDF</a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">


<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>BIOL 322 Lab 3. Genetic Drift</h1>
                        <h4 data-toc-skip class="author">Angela
Roles</h4>
            
            <h4 data-toc-skip class="date">Spring 2026</h4>
      

      <div class="d-none name"><code>biol322-lab3.Rmd</code></div>
    </div>

    
    
<p>When you have completed the exercise below, submit your model and
your lab write-up via the <a href="https://docs.google.com/forms/d/e/1FAIpQLSdA2mkUojv-37gaK0ON-B8B2a7y5cxnMUJRhaqX747QtbMxhA/viewform?usp=header" class="external-link">Google
Drive Form</a>. Responses due by the following week’s lab period.</p>
<p>During lab, we will work in groups. Here are the groups assigned for
today’s exercise:</p>
<ul>
<li>Ella and Miles</li>
<li>Chloe and Arc</li>
<li>Emmy and Natalie</li>
<li>Jordyn and Skylar</li>
<li>Amanda and Savannah</li>
<li>Clara and Paige</li>
<li>Amy and Bethany</li>
</ul>
<div class="section level2">
<h2 id="overview">Overview<a class="anchor" aria-label="anchor" href="#overview"></a>
</h2>
<p>Today, we’ll create a modified version of our Hardy-Weinberg model
from last week and we’ll add genetic drift. Think about how we describe
the effects of genetic drift on genetic variation. Starting with our
base HW model, which parameters should we adjust and what things do we
need to add to incorporate genetic drift?</p>
<p>This is somewhat complicated, since drift is a stochastic process.
Remember from Chapter 6, we know the variance in allele frequency change
due to drift; we can use that to add drift to our gene pool. “Drift”
will be a change to the allele frequency, where the value will be a
random number chosen from a normal distribution with a mean of zero and
the binomial standard deviation. In this way, we add (or subtract) a
small, random number to the allele frequency each generation. We’ll make
this change and then use the adjusted allele frequency (padj) to
randomly combine alleles to form zygotes.</p>
<p>Note that this should mean our population will always remain in HW
proportions, even though allele frequencies will be changing each
generation! This is a good demonstration of the fact that nonrandom
mating is the only thing that puts a population out of HW proportions.
For other evolutionary forces, we must examine change in allele
frequency across generations to detect the effects.</p>
</div>
<div class="section level2">
<h2 id="building-the-drift-model">Building the Drift Model<a class="anchor" aria-label="anchor" href="#building-the-drift-model"></a>
</h2>
<p>To begin, we’ll set up a model similar to the HW one, except we’ll
only keep track of 1 allele (which makes it easier to add the
stochasticity to the model without messing things up!). Here’s a
screenshot of the final product, for reference as we proceed:</p>
<div class="figure">
<img src="../reference/figures/driftmodel.png" alt="Image of the completed drift model." class="spaced-image"><p class="caption">
Image of the completed drift model.
</p>
</div>
<ol style="list-style-type: decimal">
<li>
<p>Open a new model and input the following settings:</p>
<table class="table">
<tr>
<td>
<p>Units for Time</p>
</td>
<td>
<p>Year</p>
</td>
</tr>
<tr>
<td>
<p>INIITIAL TIME =</p>
</td>
<td>
<p>0</p>
</td>
</tr>
<tr>
<td>
<p>FINAL TIME =</p>
</td>
<td>
<p>1000</p>
</td>
</tr>
<tr>
<td>
<p>Time Step</p>
</td>
<td>
<p>1</p>
</td>
</tr>
</table>
<p><br></p>
</li>
<li><p>Create a stock for each genotype, perhaps named: genoA1A1,
genoA1A2, genoA2A2. Like in the image, be sure to leave space for the
additional variables we’ll be adding!<br></p></li>
<li><p>Next, add the flows, one per genotype stock. Be sure that the
arrow on the flow connects to the stock. We name these delta11,
deltaA1A1, deltaA1A2, deltaA2A2.<br></p></li>
<li>
<p>Now we have a number of additional variables to add:</p>
<style type="text/css">
td
{
 padding:5px 8px 0px 10px;
 border: 1px solid black;
}
</style>
<table class="table">
<tr>
<td>
<p><b>Variable</b></p>
</td>
<td>
<p><b>Description</b></p>
</td>
</tr>
<tr>
<td>
<p>N</p>
</td>
<td>
<p>population size, which we’ll want to manipulate with Simulation
Control</p>
</td>
</tr>
<tr>
<td>
<p>p0</p>
</td>
<td>
<p>this is the starting frequency of the A1 allele, we’ll also vary this
with Simulation Control</p>
</td>
</tr>
<tr>
<td>
<p>freqA1A1</p>
</td>
<td>
<p>the observed frequency of the A1A1 genotype</p>
</td>
</tr>
<tr>
<td>
<p>freqA1A2</p>
</td>
<td>
<p>the observed frequency of the A1A2 genotype</p>
</td>
</tr>
<tr>
<td>
<p>freqA2A2</p>
</td>
<td>
<p>the observed frequency of the A2A2 genotype</p>
</td>
</tr>
<tr>
<td>
<p>p</p>
</td>
<td>
<p>this is the frequency of the p allele, measured every generation,
this is the primary variable we are interested in tracking to determine
effects of other variables on it</p>
</td>
</tr>
<tr>
<td>
<p>driftSD</p>
</td>
<td>
<p>the standard deviation for the distribution of changes due to drift;
this will vary depending on the population size and is used in the drift
variable</p>
</td>
</tr>
<tr>
<td>
<p>drift</p>
</td>
<td>
<p>the actual amount by which we will change the value of p for the next
generation, causing change due to drift</p>
</td>
</tr>
<tr>
<td>
<p>p adj</p>
</td>
<td>
<p>the adjusted value of p, so the new value of p after adding the
effect of drift; this is what we use to create the next generation</p>
</td>
</tr>
<tr>
<td>
<p>toggledrift</p>
</td>
<td>
<p>this variable let’s us run the model with or without causing drift to
happen; you adjust it once per run, drift is either on or off for the
whole simulation, makes it easy to see what the effect of drift is
compared to HW equilibrium</p>
</td>
</tr>
</table>
<p><br></p>
</li>
<li><p>Next, you need to add the appropriate arrows, try to follow the
image above. You can check the Arrows Key table if needed (see the
drop-down menu).<br></p></li>
<li><p>Once we have our arrows placed, we are now in position to enter
the equations for each variable. If we don’t have the arrows correct, we
won’t be able to put the right equations in, so pay careful
attention.<br></p></li>
<li><p>Remember, to enter equations, click on the Equations icon and
then click on the variable whose info you want to edit. In the box that
pops up, you enter the correct information for that variable. We have a
lot of variables so let’s take this slowly.<br></p></li>
<li><p>Let’s start with the variables that we are just setting a value
to initiate the model. For these, the equation is simple, it’s just the
starting value we are choosing to use. Use the following equations for
the varibles below:</p></li>
</ol>
<table class="table">
<tr>
<td>
<b>Variable
</b>
</td>
<td>
<b>Parameter
</b>
</td>
<td>
<b>Value
</b>
</td>
<td>
<b>Min
</b>
</td>
<td>
<b>Max
</b>
</td>
<td>
<b>Incr
</b>
</td>
</tr>
<tr>
<td>
p0
</td>
<td>
Equations
</td>
<td>
0.5
</td>
<td>
0
</td>
<td>
1
</td>
<td>
0.01
</td>
</tr>
<tr>
<td>
N
</td>
<td>
Equations
</td>
<td>
100
</td>
<td>
0
</td>
<td>
500,000
</td>
<td>
1
</td>
</tr>
</table>
<p><br></p>
<ol start="9" style="list-style-type: decimal">
<li>
<p>Having set up N and p0, we can enter the initial values for the
genotypes. Since we are assuming that mating is random, we expect
genotypes to occur according to HW proportions, so we’ll use those
equations to set the initial value for our genotypes, multiplying by N
so we get the number of individuals, not just the frequency.</p>
<ul>
<li>Make sure you do NOT change the equation entry for the genotypes,
that is already set up correctly to be the value of the flow.<br>
</li>
<li>Click on ‘genotypeA1A1’. What should you enter for the initial
value? (p0 * p0 * N)<br>
</li>
<li>For genotypeA1A2, use the equation N - genoA1A1 - genoA2A2. That
ensures that we end up with the right number of total individuals,
N.<br>
</li>
</ul>
</li>
<li>
<p>Once we have the numbers for the genotype stocks, we can set up
the equations for freq11, freq12 and freq22. We mainly want these
variables for ease of graphing and manipulation to do other
calculations. What should the equations for these variables be?</p>
<ul>
<li>Example: freq11’s equation is geno11 / N<br>
</li>
</ul>
</li>
<li><p>Now to the variable of greatest interest: p! How do we calculate
the frequency of the allele using the genotype frequencies? If you
aren’t sure, check with Angie. You can also check the key at the end
though it’s best if you can try to work it out.<br></p></li>
<li>
<p>For the variables related to drift (driftSD, seed, drift, and
toggledrift and p adj), the equations are somewhat complicated so we’ll
learn a little about how they work as we enter them into the model. Also
see the table below for quick reference.<br></p>
<ul>
<li><p><b>driftSD</b> is the standard deviation for the amount of change
due to drift. Since change due to drift is random, it could be a
positive value or a negative value but we know the amount of change
depends on the population size. In a small population, a large change is
more likely to occur than in a large population. We can control this
factor with the variable driftSD. We’re using the equation for binomial
variance here, you will see that how likely small versus large changes
are depends on the current allele frequency and the population size. The
square root part occurs because the standard deviation is the square
root of a variance.<br><br> The equation is: SQRT( (p * (1 - p) ) / (2 *
N) )<br></p></li>
<li><p><b>toggledrift</b> is the variable we will use to turn drift on
or off and to tell Vensim to create a new simulation run with drift (we
will use it as a seed in the equation for drift). This is useful so that
we know what would happen if drift was not occurring and also to allow
us to check that the model is set up correctly. It will also help us to
run multiple replicate runs in which drift can act differently – each
different value of toggledrift will correspond to a different random
outcome of the model. The value of toggledrift = 0 will be special
though, that will be what we use to run the model with no randomness at
all. We will encode this ability when we put in the equation for p adj,
which we’ll do shortly. Here’s the equation and other values to use:
<br></p></li>
</ul>
</li>
</ol>
<table class="table">
<tr>
<td>
<b>Variable
</b>
</td>
<td>
<b>Equation
</b>
</td>
<td>
<b>Min
</b>
</td>
<td>
<b>Max
</b>
</td>
<td>
<b>Incr
</b>
</td>
</tr>
<tr>
<td>
toggledrift
</td>
<td>
0
</td>
<td>
0
</td>
<td>
10000
</td>
<td>
1
</td>
</tr>
</table>
<p><br></p>
<ul>
<li>
<p><b>drift</b> is the variable we’ll use to figure out (randomly)
how much p changes by chance, calculated anew each generation. We just
set up the standard deviation (SD) for this distribution, now we use a
function (RANDOM NORMAL) to choose a value from the distribution. Change
is equally likely to be an increase as a decrease, so we’ll use a mean
of zero. For this function, we also have to specify a minimum and a
maximum so we’ll use -1 and 1 since allele frequency can’t possibly be
larger or smaller than that! The function requires arguments in the
following order RANDOM NORMAL(min, max, mean, sd, seed)<br><br> The
equation is RANDOM NORMAL(-2, 2, 0, driftSD, toggledrift)<br></p>
<ul>
<li>We’re finally ready for the most complicated equation of all, the
one for adjusting the allele frequency for the effect of drift, p adj
(p-adjust). For this variable, since we want to turn drift on and off,
we need to use an if statement – what to do when drift is on and what to
do when it’s off. We also need to make sure that when we add together p
and drift, we don’t get a value less than zero or greater than 1. If so,
we just want the value to be 0 or 1, since it’s impossible in reality to
exceed these limits. The function in Vensim that we need is IF THEN
ELSE( <em>cond</em> , <em>ontrue</em> , <em>onfalse</em> ). You can see
it requires 3 arguments (the if condition, what to do when it’s true,
and what to do when it’s false). We actually have 3 if conditions to
meet so we’re going to have a complicated nested if-then-else statement.
Here’s the equation and I’ll provide a text explanation after
it.<br><br>
</li>
</ul>
<p>IF THEN ELSE(toggledrift=0, p , IF THEN ELSE( (p+drift) &gt;= 1.0 , 1
, IF THEN ELSE( (p+drift)&lt;=0 , 0 , p+drift ) ) )<br><br></p>
<p>You can see the three if-then-else functions. Reading this from left
to right, this statement says IF toggledrift is set to be 0, then make p
adj be the same as p (this is what happens when there is no drift). But
the ELSE part says, IF (p + drift) is greater than or equal to 1.0, make
p adj equal to 1.0 (this ensures we don’t have more than 100% A1
alleles, which is impossible). And then the second ELSE says IF (p +
drift) is less than or equal to zero, set p adj to be 0 (preventing us
from having negative numbers of alleles). Finally, the last ELSE says if
none of the preceding things are true (thus, p + drift is between 0 and
1), set p adj equal to (p + drift).<br></p>
</li>
</ul>
<ol start="13" style="list-style-type: decimal">
<li><p>Ok, that was the hardest part. All we have left are the equations
for deltaA1A1, deltaA1A2, and deltaA2A2. Because we are causing drift to
happen on allele frequency directly, these equations will be similar to
what we used last week with the HW model, just using p adj instead of
p. So, for deltaA1A1, we get:<br><br> deltaA1A1 equation is: (N * p adj
* p adj) - genoA1A1<br><br> Go ahead and set up the equations for
deltaA1A2 and deltaA2A2 as well.<br></p></li>
<li>
<p>Once you have finished entering equations, check to make sure
they are correct (you may want to use the key at the end). Then we are
ready to click Simulate to run the model. To get a graph, select the
three genotype stocks (or whatever variable you want) and then click
Graph. Remember that we set up the initial conditions so that drift will
NOT occur (toggledrift = 0) so we expect to see just the HW model.</p>
<ul>
<li>What happened? Does anything change on the graphs? What did you
expect to see? Consider your responses as you write up the lab exercise
in the next section.<br>
</li>
</ul>
</li>
<li>
<p>Now, to make sure that drift can occur as expected, we need to
use Simulation Control to alter the value of toggledrift and then
Simulate the model again.<br></p>
<ul>
<li>Click Simulation Control on the first horizontal icon bar. A box
will pop up where we’ll save the changes we make. In the popup box,
change the run name to nodrift and in the Comment box type ‘toggledrift
= 0’, this will be our first run where we don’t change the value of
toggledrift. Click Save Changes and then click Simulate.<br>
</li>
<li>Now we want to create a new simulation with toggledrift = 1.0 so
that drift will occur. Click on Simulation Control again and this time
change the run name to run1 and in Comments type ‘toggledrift = 1’.
Click on the variable toggledrift and change the value to 1. Then click
Save Changes and Simulate.<br>
</li>
<li>Let’s make one more simulation with drift. Click on Simulation
Control and name this one run2, in Comments: toggledrift = 2, and
finally click on toggledrift and change the value to 2. Click Save
Changes and then Simulate.<br>
</li>
</ul>
</li>
<li>
<p>Once you have created those variations, click Simulate on the
horizontal menu bar, click on p, and then click Graph. A graph should
appear that has a line for each value of toggledrift that you tested.
What happened? How might we explain what’s going on?<br><br></p>
<pre><code>Note that for the drift model, we are primarily interested in what happens to the allele frequency, p, over time. So instead of graphing genotype frequencies, we want to graph allele frequency. We could also graph heterozygosity, as that's our common estimate of genetic variation.&lt;br&gt;&lt;br&gt;</code></pre>
<p><i>(Sidebar: you should see a flat line for the nodrift run and then
two squiggly lines that are different from each other for run1 and
run2.)</i></p>
</li>
</ol>
</div>
<div class="section level2">
<h2 id="lab-write-up">Lab Write-Up<a class="anchor" aria-label="anchor" href="#lab-write-up"></a>
</h2>
<p>Now that you have created an awesome working drift model, let’s see
what happens with multiple runs and altering starting conditions.
Remember that since drift is a stochastic process, we cannot predict the
outcome of any particular run but we can see trends from multiple runs.
Thus, we’ll need to look at multiple runs of any conditions we are
testing out to identify trends. Remember that the effect of drift
depends on both initial allele frequencies and total population size (a
larger population takes longer to reach equilibrium).</p>
<p>For the questions below, you should pay attention to the following
values for each run so that you can compare across different runs. Which
variables change the outcome and in what ways? Does the time to fixation
change? Does the size of change from one generation to the next
change?</p>
<ul>
<li>population size
<ul>
<li>length of run (how many time steps before p = 0 or 1; time to allele
extinction or fixation)</li>
<li>where allele frequency started and where it ended up (graph p)</li>
<li>where heterozygosity started and where it ended up (graph freqA1A2
to see this)</li>
</ul>
</li>
</ul>
<p><b>So you will be graphing p and possibly freqA1A2
(heterozygosity).</b> Examine the trends associated with these variables
when you run the scenarios below. Before running them, try to predict
what you think will happen and which variables will matter the most.
Consider how the differences between scenarios demonstrates the way in
which drift influences genetic diversity.</p>
<div class="section level3">
<h3 id="questions">Questions<a class="anchor" aria-label="anchor" href="#questions"></a>
</h3>
<p>1, <b>Explore the effect of initial allele frequency on the outcome
of the runs.</b> Do 10 runs for each of the following sets of starting
conditions:</p>
<table class="table">
<tr>
<td>
Trial
</td>
<td>
p0
</td>
<td>
N
</td>
<td>
Final Time
</td>
<td>
toggledrift
</td>
<td>
# runs
</td>
</tr>
<tr>
<td>
1a
</td>
<td>
0.5
</td>
<td>
100
</td>
<td>
1000
</td>
<td>
1 to 10
</td>
<td>
10
</td>
</tr>
<tr>
<td>
1b
</td>
<td>
0.2
</td>
<td>
100
</td>
<td>
1000
</td>
<td>
1 to 10
</td>
<td>
10
</td>
</tr>
<tr>
<td>
1c
</td>
<td>
0.8
</td>
<td>
100
</td>
<td>
1000
</td>
<td>
1 to 10
</td>
<td>
10
</td>
</tr>
</table>
<p><br></p>
<p>The easiest way to do this is to change the default values for p0, N,
and Final Time then use Simulation Control to just change the value of
toggledrift 10 times. <b>You want 1 graph of p showing all 10 runs of
trial 1a</b>, a separate graph of the run for trial 1b and for trial 1c.
You can change the Final Time by clicking on the Model drop-down menu
and choosing Settings… I also suggest saving your graph for each trial.
To save a graph, go to File –&gt; Save. You can also rescale and zoom in
or out by clicking on the graph (use the View menu to reset the
scale).<br><br></p>
<p>Now, tell me what you notice about what happens in these runs. Is
equilibrium reached? Fixation or extinction? What changes when you
change starting allele frequency? Write a summary of your
observations.<br></p>
<ol start="2" style="list-style-type: decimal">
<li>
<b>Examine what happens when you change population size.</b> Use the
following sets of starting conditions:</li>
</ol>
<table class="table">
<tr>
<td>
Trial
</td>
<td>
p0
</td>
<td>
N
</td>
<td>
Final Time
</td>
<td>
toggledrift
</td>
<td>
# runs
</td>
</tr>
<tr>
<td>
2a
</td>
<td>
0.3
</td>
<td>
10
</td>
<td>
1000
</td>
<td>
1 to 10
</td>
<td>
10
</td>
</tr>
<tr>
<td>
2b
</td>
<td>
0.3
</td>
<td>
100
</td>
<td>
1000
</td>
<td>
1 to 10
</td>
<td>
10
</td>
</tr>
<tr>
<td>
2c
</td>
<td>
0.3
</td>
<td>
1000
</td>
<td>
1000
</td>
<td>
1 to 10
</td>
<td>
10
</td>
</tr>
</table>
<p><br></p>
<p>Again, you should do at least 10 runs for each value of population
size so that you can try to detect any trends. What trends do you
notice? How does population size influence the equilibrium that’s
reached? Feel free to explore what happens with other values of N or
with other values for p0. Write up a summary of your
observations.<br></p>
<ol start="3" style="list-style-type: decimal">
<li>Finally, in paragraph form, give me a summary of the overall
ideas/patterns demonstrated in the drift model (you don’t need to repeat
yourself from earlier questions). Reflect on what you’ve observed. Are
any of the results surprising to you? Or do they change how you think
about the consequences of drift? How might these results influence
thinking about drift as a potential issue in conservation? (You are, of
course, welcome to test out other conditions in your model!)</li>
</ol>
<p><b>Find your saved model file (should end in .mdl). Submit the model
and your responses to the <a href="https://docs.google.com/forms/d/e/1FAIpQLSdA2mkUojv-37gaK0ON-B8B2a7y5cxnMUJRhaqX747QtbMxhA/viewform?usp=header" class="external-link">Google
Drive Form</a>. Write-up are due by the next lab period.</b><br></p>
</div>
</div>
<div class="section level2">
<h2 id="key-to-arrows">Key to Arrows<a class="anchor" aria-label="anchor" href="#key-to-arrows"></a>
</h2>
Below find the key to arrow placements for each variable.
<table style="width:95%" class="table">
<tr>
<td style="width:10%">
<b>Variable
</b>
</td>
<td style="width:50%">
<b>Arrow pointing to
</b>
</td>
<td style="width:30%">
<b>Arrow arriving from
</b>
</td>
</tr>
<tr>
<td>
genoA1A1
</td>
<td>
freqA1A1; deltaA1A1; genoA1A2
</td>
<td>
p0; N
</td>
</tr>
<tr>
<td>
genoA1A2
</td>
<td>
freqA1A2; deltaA1A2
</td>
<td>
N; genoA1A1; genoA2A2
</td>
</tr>
<tr>
<td>
genoA2A2
</td>
<td>
freqA2A2; deltaA2A2; genoA1A2
</td>
<td>
p0; N
</td>
</tr>
<tr>
<td>
N
</td>
<td>
genoA1A1; genoA1A2; genoA2A2; freqA1A1; freqA1A2; freqA2A2; deltaA1A1;
deltaA1A2, deltaA2A2; driftSD
</td>
<td>
none
</td>
</tr>
<tr>
<td>
freqA1A1
</td>
<td>
p
</td>
<td>
genoA1A1; N
</td>
</tr>
<tr>
<td>
freqA1A2
</td>
<td>
p
</td>
<td>
genoA1A2; N
</td>
</tr>
<tr>
<td>
freqA2A2
</td>
<td>
none
</td>
<td>
genoA2A2; N
</td>
</tr>
<tr>
<td>
p
</td>
<td>
freqA1A1; freqA1A2
</td>
<td>
p adj; driftSD
</td>
</tr>
<tr>
<td>
driftSD
</td>
<td>
drift
</td>
<td>
N; p
</td>
</tr>
<tr>
<td>
toggledrift
</td>
<td>
none
</td>
<td>
p adj; drift
</td>
</tr>
<tr>
<td>
drift
</td>
<td>
p adj
</td>
<td>
driftSD; toggledrift
</td>
</tr>
<tr>
<td>
p adj
</td>
<td>
deltaA1A1; deltaA1A2; deltaA2A2
</td>
<td>
p; drift; toggledrift
</td>
</tr>
<tr>
<td>
deltaA1A1
</td>
<td>
none
</td>
<td>
p adj; N; genoA1A1
</td>
</tr>
<tr>
<td>
deltaA1A2
</td>
<td>
none
</td>
<td>
p adj; N; genoA1A2
</td>
</tr>
<tr>
<td>
deltaA2A2
</td>
<td>
none
</td>
<td>
p adj; N; genoA2A2
</td>
</tr>
</table>
<p><br></p>
<a class="anchor" id="eqnkey"></a>
<section><h2>
Key to Equations
</h2>
Below find the table of equations for each variable.
<table style="width:95%" class="table">
<tr>
<td style="width:5%">
<b>Variable
</b>
</td>
<td style="width:5%">
<b>Parameter
</b>
</td>
<td style="width:80%">
<b>Value of equations or initial
</b>
</td>
</tr>
<tr>
<td>
N
</td>
<td>
Equations
</td>
<td>
100
</td>
</tr>
<tr>
<td>
p0
</td>
<td>
Equations
</td>
<td>
0.5
</td>
</tr>
<tr>
<td>
genoA1A1
</td>
<td>
Initial
</td>
<td>
p0 * p0 * N
</td>
</tr>
<tr>
<td>
genoA1A2
</td>
<td>
Initial
</td>
<td>
N - genoA1A1 - genoA2A2
</td>
</tr>
<tr>
<td>
genoA2A2
</td>
<td>
Initial
</td>
<td>
(1 - p0) * (1 - p0) * N
</td>
</tr>
<td>
freqA1A1
</td>
<td>
Equations
</td>
<td>
genoA1A1 / N
</td>

<tr>
<td>
freqA1A2
</td>
<td>
Equations
</td>
<td>
genoA1A2 / N
</td>
</tr>
<tr>
<td>
freqA2A2
</td>
<td>
Equations
</td>
<td>
genoA2A2 / N
</td>
</tr>
<tr>
<td>
p
</td>
<td>
Equations
</td>
<td>
freqA1A1 + 0.5 * freqA1A2
</td>
</tr>
<tr>
<td>
driftSD
</td>
<td>
Equations
</td>
<td>
SQRT( (p * (1 - p) ) / (2 * N) )
</td>
</tr>
<tr>
<td>
toggledrift
</td>
<td>
Equations
</td>
<td>
0
</td>
</tr>
<tr>
<td>
drift
</td>
<td>
Equations
</td>
<td>
RANDOM NORMAL( -2, 2, 0, driftSD, toggledrift)
</td>
</tr>
<tr>
<td>
p adj
</td>
<td>
Equations
</td>
<td>
IF THEN ELSE(toggledrift = 0, p , IF THEN ELSE( (p + drift) &gt;= 1 , 1
, IF THEN ELSE( (p + drift) &lt;= 0 , 0 , p + drift ) ) )
</td>
</tr>
<tr>
<td>
deltaA1A1
</td>
<td>
Equations
</td>
<td>
(p adj * p adj * N) - genoA1A1
</td>
</tr>
<tr>
<td>
deltaA1A2
</td>
<td>
Equations
</td>
<td>
(2 * p adj * (1 - p adj) * N) - genoA1A2
</td>
</tr>
<tr>
<td>
deltaA2A2
</td>
<td>
Equations
</td>
<td>
( (1 - p adj) * (1 - p adj) * N) - genoA2A2
</td>
</tr>
</table>
<p><br></p>
</section>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Prof. A. Roles.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer>
</div>





  </body>
</html>
